Extension/src/githubAPI.ts

```diff
@@ -172,19 +184,25 @@ export async function getTargetBuildInfo(updateChannel: string): Promise<BuildIn
  * @param updateChannel The user's updateChannel setting.
  * @return The Build if the user should update to it, otherwise undefined.
  */
-function getTargetBuild(builds: Build[], userVersion: PackageVersion, updateChannel: string): Build | undefined {
+export function getTargetBuild(builds: Build[], userVersion: PackageVersion, updateChannel: string): Build | undefined {
     if (!vscode.workspace.getConfiguration("extensions", null).get<boolean>("autoUpdate")) {
         return undefined;
     }
+    const latestVersion: PackageVersion = new PackageVersion(builds[0].name);
+    if ((updateChannel === 'Insiders') && (userVersion.suffix && userVersion.suffix === 'insiders') && (userVersion.isGreaterThan(latestVersion))) {
```

Comment:
Do you need to check the suffix? I think just using `isGreaterThan` is better. You don't want to accidentally downgrade from 0.27.0 to 0.27.0-insiders (which would also be an internal testing case which I think was missing from the table)

Please also add a comment that explains why we do this here instead of as part of `useBuild` (so we don't have to check every build in the `builds` array).